openai/openai.go

```diff
@@ -70,65 +104,40 @@ func (oac *OpenAiClient) GetCompletion(prompt string, debug bool) (*string, erro
 		fmt.Print(logMsg)
 	}
 
-	resp, err := oac.client.CreateChatCompletion(
-		context.Background(),
-		openai.ChatCompletionRequest{
```

Comment:
The RequestCompletion function could be refactored into smaller methods for readability and to handle different responsibilities.

Reply:
lol you really won't let go of this will you?